posthog/hogql/database/schema/util/where_clause_extractor.py

```diff
@@ -47,107 +41,168 @@ class SessionMinTimestampWhereClauseExtractor(CloningVisitor):
     context: HogQLContext
     clear_types: bool = False
     clear_locations: bool = False
+    capture_timestamp_comparisons: bool = False
+    our_tables: list[ast.LazyTable | ast.LazyJoin]
+    tombstone_string: str
+
+    limits: Literal["", "session_min"] = ""
 
-    def get_inner_where(self, parsed_query: ast.SelectQuery) -> Optional[ast.Expr]:
-        if not parsed_query.where:
+    def __init__(self, context: HogQLContext):
+        super().__init__()
+        self.context = context
+        self.our_tables = []
+        # A constant with this string will be used to escape early if we can't handle the query
+        self.tombstone_string = (
+            "__TOMBSTONE__" + ("".join(random.choices(string.ascii_uppercase + string.digits, k=10))) + "__"
+        )
+
+    def add_local_tables(self, join_or_table: LazyJoinToAdd | LazyTableToAdd):
+        """Add the tables whose filters to extract into a new where clause."""
+        if isinstance(join_or_table, LazyJoinToAdd):
+            if join_or_table.lazy_join not in self.our_tables:
+                self.our_tables.append(join_or_table.lazy_join)
+        elif isinstance(join_or_table, LazyTableToAdd):
+            if join_or_table.lazy_table not in self.our_tables:
+                self.our_tables.append(join_or_table.lazy_table)
+
+    def get_inner_where(self, select_query: ast.SelectQuery) -> Optional[ast.Expr]:
+        """Return the where clause that should be applied to the inner table. If None is returned, no pre-filtering is possible."""
+        if not select_query.where and not select_query.prewhere:
             return None
 
         # visit the where clause
-        where = self.visit(parsed_query.where)
+        wheres = []
+        if select_query.where:
+            wheres.append(select_query.where)
+        if select_query.prewhere:
+            wheres.append(select_query.prewhere)
+
+        if len(wheres) == 1:
+            where = self.visit(wheres[0])
+        else:
+            where = self.visit(ast.And(exprs=wheres))
 
         if isinstance(where, ast.Constant):
             return None
 
         return clone_expr(where, clear_types=True, clear_locations=True)
 
     def visit_compare_operation(self, node: ast.CompareOperation) -> ast.Expr:
-        is_left_constant = is_time_or_interval_constant(node.left)
-        is_right_constant = is_time_or_interval_constant(node.right)
-        is_left_timestamp_field = is_simple_timestamp_field_expression(node.left, self.context)
-        is_right_timestamp_field = is_simple_timestamp_field_expression(node.right, self.context)
+        if has_tombstone(node, self.tombstone_string):
+            return ast.Constant(value=self.tombstone_string)
```

Comment:
Does this not break parsing/resolving? Going from `WHERE column_a = 'some_value'` to `WHERE '__tombstone__'`?

Reply:
Not really, as in `get_inner_where`, the meat of the extraction, we have:

```python
        if isinstance(where, ast.Constant):
            return None
```